functions/go/set-labels/transformer/labels.go

```diff
@@ -0,0 +1,164 @@
+package transformer
+
+import (
+	"encoding/json"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
+)
+
+type FieldPath []string
+type GVK struct {
+	group   string
+	version string
+	kind    string
+}
+
+// the similar struct esxits in resid.GVK, but there is no function to create an GVK struct without using kyaml
+type FieldSpec struct {
+	Identifier         GVK
+	Path               FieldPath
+	CreateIfNotPresent bool
+	//TODO: should support user configurable field for labels
+}
+
+type FieldSpecs []FieldSpec
+
+type LabelTransformer struct {
+	// Desired labels
+	NewLabels map[string]string
+	// FieldSpecs storing default label fields
+	FieldSpecs []FieldSpec
+	// Results is used to track labels that have been applied
+	Results fn.Results
+}
+
+// perform the whole set labels operation according to given resourcelist
+func SetLabels(rl *fn.ResourceList) (bool, error) {
```

Comment:
nit: godoc shall start with the function name so this docstring need to be `// SetLabels performs ...`

Reply:
Here's the guidance https://go.dev/blog/godoc